server/src/main/java/io/druid/client/indexing/ClientCompactQueryTuningConfig.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.client.indexing;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.druid.segment.IndexSpec;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public class ClientCompactQueryTuningConfig
+{
+  // These default values should be synchronized with those of IndexTuningConfig
```

Comment:
I think you meant `RealtimeTuningConfig`. However, in that case, we should probably declare these as `public static final xxx` in one of these classes and other class should refer to those. then they would stay same automatically.

Reply:
A compaction task is essentially an indexTask + ingestSegmentFirehose. This class should be  synchronized with IndexTuningConfig.
Unfortunately, ```druid-server``` currently has no dependency on ```druid-indexing-service```, so it cannot use the IndexTuningConfig for now. This is why [this discussion](https://github.com/druid-io/druid/pull/5102#discussion_r158430657) comes up.

Reply:
yeah ,  I know ... its unfortunate to have all these classes in `server/` with counterparts in `indexing-service/` . we should probably think about merging them.
maybe the solution is to merge everything from `indexing-service/` into `server/`

created https://github.com/druid-io/druid/issues/5229 to trach it.